teachertool/src/components/CriteriaInstanceDisplay.tsx

```diff
@@ -22,22 +23,32 @@ const InlineInputSegment: React.FC<InlineInputSegmentProps> = ({
     shouldExpand,
     numeric,
 }) => {
-    function onChange(newValue: string) {
+    const [isEmpty, setIsEmpty] = useState(!initialValue);
+
+    function onDebouncedChange(newValue: string) {
         setParameterValue(instance.instanceId, param.name, newValue);
     }
 
+    function onChange(newValue: string) {
+        // Keep this out of the debounced version to avoid delayed appearance.
+        setIsEmpty(!newValue);
+    }
+
     return (
         <DebouncedInput
             className={classList(
                 css["inline-input"],
                 numeric ? css["number-input"] : css["string-input"],
                 shouldExpand ? css["long"] : undefined,
+                isEmpty ? css["error"] : undefined
             )}
+            icon={isEmpty ? "fas fa-exclamation-triangle" : undefined}
             initialValue={initialValue}
+            onDebouncedChange={onDebouncedChange}
             onChange={onChange}
             preserveValueOnBlur={true}
             placeholder={numeric ? "0" : param.name}
-            title={param.name}
+            title={isEmpty ? lf("{0}: value required", param.name) : param.name}
```

Comment:
The hit box for the input is a bit weird when there's an error.. If you hover over the placeholder, you get this tooltip, but if you hover over the icon, you get the tooltip associated with the criteria instance.

Reply:
Fixed. At first, I tried just setting iconTitle but that didn't work as expected when it's not a button, and I was a little worried about messing up existing scenarios if I changed that, so I just wrapped it in a div with the desired tooltip.

With fix: https://makecode.microbit.org/app/6d8a16dc031a6cba000e6c00781331f5a6da05cb-d358cf977b--eval?tc=1